app/models/submission_state_actions.rb

```diff
@@ -12,12 +12,16 @@ def initialize(submission)
   end
 
   def run
-    actions = []
-    actions << approve_action if submission.submitted?
-    actions << publish_action if submission.submitted? || submission.approved?
-    actions << reject_action if submission.submitted?
-    actions << close_action if submission.submitted? || submission.approved?
-    actions
+    case submission.state
```

Comment:
❤️ ❤️ ❤️

Reply:
Look at that it's like you're a Ruby programmer now!!

Reply:
❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️